### PR TITLE
feat(docs): add community extensions page

### LINKED ISSUE #5702: [Docs] Add a Community extension page
Add a community extension page to list extensions built by the community. It will be similar to the community connector page: https://loopback.io/doc/en/lb4/Community-connectors.html. 

## Acceptance criteria
- [ ] add a community extension page with a few community extensions we know about. 
    - the idea is to have the community user to submit PRs to add their extensions on this page
- [ ] add this to sidebar of the docs
- [ ] Add link to npm keyword "loopback4-extension" page.
- [ ] Add links to pages about creating extensions.

cc @raymondfeng @bajtos, per our discussion today.

### PR DESCRIPTION
**Pull request by dhmlau**
Fixes #5702

Add a community extensions page with some of the extensions we know about. The intention of this page is to have community adding their own extensions there. 

Preview of the change: 
Screen Shot 2020-06-09 at 11 05 24 PM

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by dhmlau** [comment]
cc @samarpanB

**Comment by dhmlau** [comment]
@slnode test please

File: docs/site/Community-extensions.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <tr>
+      <td><a href="https://www.npmjs.com/package/loopback-component-crud">loopback-component-crud</a></td>
+      <td>Generic CRUD controller mixin, supports authentication, nested authorization(cascade)</td>
+    </tr>
+    <tr>
+      <td><a href="https://www.npmjs.com/package/loopback-component-authorization">loopback-component-authorization</a></td>
+      <td>Implement HRBAC authorization</td>
+    </tr>
+  </tbody>
+<table>
```
Comment (bajtos) [inline docs/site/Community-extensions.md:92]: Please add instructions on how can extension authors contribute their extensions to our list. Feel free to open a follow-up pull request for that.

You can inspire in our existing docs for contributing LB3 extensions: https://loopback.io/doc/en/community/How-to-add-a-project.html

File: docs/site/Community-extensions.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  <thead>
+    <tr>
+      <th width="160">Extension</th>
+      <th width="280">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><a href="https://www.npmjs.com/package/loopback4-soft-delete">loopback4-soft-delete</a></td>
+      <td>Used for soft-deletes in a multi-tenant application</td>
```
Comment (samarpanB) [inline docs/site/Community-extensions.md:36]: This extension is generic not specifically for multi tenant application. Can you reword that please?

Comment (dhmlau) [reply docs/site/Community-extensions.md:36]: do you have a suggestion?

Comment (samarpanB) [reply docs/site/Community-extensions.md:36]: An extension for adding soft-delete feature to your models and repositories

**Comment by bajtos** [comment]
On the second thought and in the light of #5718, I am not sure if it is a good idea to have a page listing community components. As a LoopBack user looking for a solution to my problem, I would not know that I have to look at the list of community extensions.

I think a better option is to treat community extensions as first-class problem solutions and list them alongside the "official" extensions, using the README-sourcing mechanism we already have in place (see [README layout](https://loopback.io/doc/en/contrib/readmes.html)). We can introduce a front-matter flag to mark an extension as community-maintained and update our README layout template to add a small notice at the top of the page to explain that the extension is not maintained by us.

**Comment by dhmlau** [comment]
Sure. I'm good with either way.

**Comment by raymondfeng** [comment]
> I think a better option is to treat community extensions as first-class problem solutions and list them alongside the "official" extensions, using the README-sourcing mechanism we already have in place (see README layout). We can introduce a front-matter flag to mark an extension as community-maintained and update our README layout template to add a small notice at the top of the page to explain that the extension is not maintained by us.

It might be beyond technical perspectives. From the legal/IP angle, I don't think we have the clearance to embed community work on `loopback.io` without CLA in place.

**Comment by dhmlau** [comment]
> It might be beyond technical perspectives. From the legal/IP angle, I don't think we have the clearance to embed community work on loopback.io without CLA in place.

That's a good point. Thanks @raymondfeng for the reminder. 
@bajtos, in this case, are you good to land this PR in the current form? If yes, I'll take care of the conflicting changes in the sidebar file. Thanks.

**Comment by samarpanB** [comment]
@dhmlau is this PR still ON or it has been put on hold ?

**Comment by dhmlau** [comment]
@bajtos, are you good to land this PR in the current form (with resolving conflicting files)? Thanks.

**Comment by dhmlau** [comment]
Since there's restructuring of the sidebar, it might be the best to get reviews from the team again. Thanks. 
cc @bajtos @jannyHou

File: docs/site/Community-extensions.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+In addition to the extensions that LoopBack provides and maintains, there are a
+number of extensions created by the open source community.
+
+To create a LoopBack extension, see
+[Extending LoopBack 4 documentation page](https://loopback.io/doc/en/lb4/Extending-LoopBack-4.html)
+for details. It is recommended to use the
+[extension generator](https://loopback.io/doc/en/lb4/Extension-generator.html)
+to scaffold a new extension.
+
+The following table lists some of the community connectors. See
```
Comment (jannyHou) [inline docs/site/Community-extensions.md:18]:
```suggestion
The following table lists some of the community extensions. See
```
?

**Comment by bajtos** [comment]
> are you good to land this PR in the current form (with resolving conflicting files)? Thanks.

I am fine to add a new page for community extensions based on the discussion above.

IIUC, you are proposing to add the new page "Community extensions" to the "How-to guides" section of the sidebar. I feel that's not the best location - personally, I would not look for community extensions there.

I am proposing to convert "Community extensions" into a first-level entry and place it between "Reference guides" and "Extending LoopBack 4" (of further down in the list).

File: docs/site/Community-extensions.md (hunk 3 above)
Comment (dhmlau) [reply docs/site/Community-extensions.md:18]: oops. typo. Thanks @jannyHou.

**Comment by dhmlau** [comment]
@bajtos, I moved the community extension page _after_ the `Extending LoopBack 4` section. It seems to be a more natural flow to me. PTAL. Thanks.

**Pull request merged**